Neos.Flow/Classes/Configuration/ConfigurationSource/AppendConfigurationSource.php

```diff
@@ -0,0 +1,53 @@
+<?php
+namespace Neos\Flow\Configuration\ConfigurationSource;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Configuration\Source\YamlSource;
+use Neos\Flow\Core\ApplicationContext;
+
+class AppendConfigurationSource implements ConfigurationSourceInterface
+{
+
+    /**
+     * @var YamlSource
+     */
+    private $yamlSource;
+
+    /**
+     * @var string
+     */
+    private $filePrefix;
+
+    public function __construct(YamlSource $yamlSource, string $filePrefix)
```

Comment:
Should we really bind those constructors to `YamlSource`? What about the possibility to load other formats?

Reply:
Since Yaml is the only configuration format we support in Flow, I say that it's okay

Reply:
Was thinking of the "vision" of this PR in the PR message that states `"and custom configuration format's. E.g. it is possible to load json files or NodeTypes"`

Reply:
Got ya'! +1 for the vision 👍 
But these core configuration sources, don't they always follow the core, so it will not change unless the core changes it's configuration format?

Do you think about a interface instead that could be implemented or did you have any other concrete idea in mind?

Reply:
yes, technically it's not something that is gonna break, because we have those constructors in the core. But we also set an example with those implementations, so it's likely that custom implementations will copy this constructor signature. And last but not least, some people will possibly extend those implementations (unless we `final` them, which we could indeed do).

Reply:
Sorry for the late response. The implemented ConfigurationSources already implement a `ConfigurationSourceInterface`. You only need to implement this if you want to provide a complete custom source without yaml.

Reply:
> But what now is a ConfigurationSource is really more of a ConfigurationProcessor, no? MergeConfigurationProecessor, AppendConfigurationProcessor etc. etc.

Yeah, you could call it a processor, but I also do think that Loader works well now. It does load config data from a source in a specific order and merge strategy.

> but I also feel that could be offloaded into an (what I would call) ConfigurationSourceInterface, that just wraps the loading and returns the loaded array so if you want to have something else than YAML you just need to implement that

By "wrap the loading" do you mean wrap `YamlSource`? Not sure this additional layer of abstraction would carry much benefit - right now, that's basically what `YamlSource` is doing with it's `load()` method. So if we just make this an interface, the same flexibility would be provided (and we can do this in a separate PR as you mentioned)